### PR TITLE
scripts: add debug build to fuzz.sh

### PR DESCRIPTION
**Pull request by cujomalainey**
Add flag for adding symbols in easily.

### CODE REVIEW

File: scripts/fuzz.sh [hunk 1]
```diff
@@ -116,6 +120,14 @@ main()
     -DCONFIG_ASAN=y
   )
 
+  local debug_configs=(
+    -DCONFIG_DEBUG=y
+    -DCONFIG_DEBUG_OPTIMIZATIONS=y
+  )
+
+  # append debug flags if a debug build
+  if $DEBUG_BUILD; then fuzz_configs=( "${fuzz_configs[@]}" "${debug_configs[@]}" ); fi
```
Comment (marc-hb) [inline scripts/fuzz.sh:129]:
```suggestion
  if $DEBUG_BUILD; then fuzz_configs+=(  "${debug_configs[@]}" ); fi
```

File: scripts/fuzz.sh [hunk 2]
```diff
@@ -17,6 +17,8 @@ Usage:
              relatively verbose stderr is not redirected by -o.
   -j n       Number of concurrent -jobs=n. Defaults to 1.
              The value 0 uses the output of the 'nproc' command.
+  -d         Add debug optimizations and symbols to the build.
+             Useful for testcase reproduction.
```
Comment (marc-hb) [inline scripts/fuzz.sh:21]: Why not add these CONFIG_DEBUG* always? Are they slowing anything down?

**Review by marc-hb — DISMISSED** [review]
This and #9343 made me realize we have a bit of a `.config` mess which I just described in brand new:
- #9386 

Please let me first clean-up this `fuzz.sh` script a little bit more accordingly. I will try to submit something tonight.

**Comment by marc-hb** [comment]
>  I will try to submit something tonight.

Done, submitted #9389.

This moves all CONFIG_ values to .conf files which will help re-use and reduce duplicate/diverge.

Based on it, could this new -d option include the already existing app/debug_overlay.conf file?

**Review by kv2019i — APPROVED** [review]

**Comment by cujomalainey** [comment]
> > I will try to submit something tonight.
> 
> Done, submitted #9389.
> 
> This moves all CONFIG_ values to .conf files which will help re-use and reduce duplicate/diverge.
> 
> Based on it, could this new -d option include the already existing app/debug_overlay.conf file?

That will be helpful also for enabling multiple sanitizers again. Much appreciated.

File: scripts/fuzz.sh [hunk 3]
```diff
@@ -116,6 +120,14 @@ main()
     -DCONFIG_ASAN=y
   )
 
+  local debug_configs=(
```
Comment (marc-hb) [inline scripts/fuzz.sh:123]: It's pretty amazing that this does NOT git conflict considering I removed the entire `fuzz_configs` variable above (and more) in merged commit 577c9ae3e75 and replaced it with `local conf_files_list=...` which looks entirely different! Yet neither `git cherry-pick` nor `git merge` seem to bother - wow!

Even if not a git conflict, this is still very much a logical conflict so this PR should not be merged as is right now.

There will likely be another conflict with incoming #9409, so maybe wait for that to be merged too.

**Comment by cujomalainey** [comment]
> @cujomalainey still needed ?

Yes but in a different form. We have symbols already. They just only work with lldb. I need to add a flag to switch to -Og from -O2.

**Comment by cujomalainey** [comment]
Closing as I think this isn't needed anymore

**Pull request closed without merging**